packages/core/useFetch/index.ts

```diff
@@ -361,9 +362,9 @@ export function useFetch<T>(url: MaybeRef<string>, ...args: any[]): UseFetchRetu
 
   function setType(type: DataType) {
     return () => {
-      if (!initialized) {
+      if (!isFetching.value) {
```

Comment:
I think this should still be `!initalized`, right? Changing the payload seems fine to me, but we don't want to change the type.

Reply:
In this PR `initialized` is not used. It was set to true the first time `execute` was called and stayed so.
Now in master now you can do

```js
const {post} = useFetch(..., {immediate: false})
// somewhere else
post(...).json().execute() // of course only once
```
If you'll allow more calls `json` will return `undefined` and you cannot chain `execute` and ... (you got the idea).

In the last commit I changed the types so you can do
```js
const {post} = useFetch(..., {immediate: false}).json()
// somewhere else
post(...).execute() // even more times
```

### To note
This is a breaking change as now the order you chain the method is strict `post -> json (method --> type)`, here instead is the inverse `json --> (post X times)`.
If the order shouldn't be strict then those functions should return the same interface, otherwise keeping track what was called and should be returned would be unnecessary